app/jobs/titre_identite_watermark_job.rb

```diff
@@ -10,77 +10,17 @@ class FileNotScannedYetError < StandardError
   # (to avoid modifying the file while it is being scanned).
   retry_on FileNotScannedYetError, wait: :exponentially_longer, attempts: 10
 
-  MAX_IMAGE_SIZE = 1500
-  SCALE = 0.9
-  WATERMARK = URI.parse(WATERMARK_FILE).is_a?(URI::HTTP) ? WATERMARK_FILE : Rails.root.join("app/assets/images/#{WATERMARK_FILE}")
-
   def perform(blob)
     return if blob.watermark_done?
     raise FileNotScannedYetError if blob.virus_scanner.pending?
 
     blob.open do |file|
-      watermark = resize_watermark(file)
-
-      if watermark.present?
-        processed = watermark_image(file, watermark)
-
-        blob.upload(processed)
-        blob.touch(:watermarked_at)
-      end
-    end
-  end
-
-  private
-
-  def watermark_image(file, watermark)
-    ImageProcessing::MiniMagick
-      .source(file)
-      .convert("png")
-      .resize_to_limit(MAX_IMAGE_SIZE, MAX_IMAGE_SIZE)
-      .composite(watermark, mode: "over", gravity: "center")
-      .call
-  end
-
-  def resize_watermark(file)
-    metadata = image_metadata(file)
+      processed = WatermarkService.new.process(file)
```

Comment:
Dans `WatermakService`, tu utilises `Tempfile.new`. Or il est écrit dans la (doc)[https://rubyapi.org/3.2/o/tempfile#method-c-new] :

```
If possible, consider instead using [Tempfile.create](https://rubyapi.org/3.2/o/tempfile#method-c-create), which:

- Avoids the performance cost of delegation, incurred when [Tempfile.new](https://rubyapi.org/3.2/o/tempfile#method-c-new) calls its superclass DelegateClass(File).
- Does not rely on a finalizer to close and unlink the file, which can be unreliable.
 ```

Du coup, je suggère que le service watermak ne s'occupe pas de la création du fichier temporaire, on délègue à l'appelant.

```suggestion
      Tempfile.create(["watermarked", file.extname]) do |output|
        processed = WatermarkService.new.process(file, output)
        return if processed.blank?
        blob.upload(processed)
        blob.touch(:watermarked_at)
      end
      
```